File: problems/Sparse Matrix Multiplication/multiply.cs
public class Solution {
    public int[][] Multiply(int[][] A, int[][] B) {
        int n = A.Length;
        int m = B[0].Length;

        var res = new int[n][];

        for (int i = 0; i < n; i++) {
            res[i] = new int[m];
            for (int j = 0; j < m; j++){
                for (int k = 0; k < B.Length; k++){
                    if (A[i][k] != 0) {
                        res[i][j] += A[i][k] * B[k][j];
                    }
                }
            }

        }

        return res;
    }
}

File: problems/Sparse Matrix Multiplication/multiply.java
class Solution {
    public int[][] multiply(int[][] mat1, int[][] mat2) {
        int[][] result = new int[mat1.length][mat2[0].length];

        for (int i = 0; mat1.length > i; ++i) {
            for (int k = 0; mat1[0].length > k; ++k) {
                if (0 != mat1[i][k]) {
                    for (int j = 0; mat2[0].length > j; ++j) {
                        if (0 != mat2[k][j]) {
                            result[i][j] += mat1[i][k] * mat2[k][j];
                        }
                    }
                }
            }
        }

        return result;
    }
}
